pycomlink/processing/k_R_relation.py

```diff
@@ -183,32 +185,70 @@ def a_b(f_GHz, pol, approx_type="ITU_2005"):
         prediction methods", International Telecommunication Union, P.838-2 (04/2003) P.838-3 (03/2005)
 
     """
-    from scipy.interpolate import interp1d
+    if isinstance(f_GHz, xr.DataArray):
+        return_xarray = True
+        f_GHz_coords = f_GHz.coords
+    else:
+        return_xarray = False
+
+    f_GHz = xr.DataArray(f_GHz)
+
+    if isinstance(pol, str):
+        pol = xr.full_like(f_GHz, pol, dtype=object)
+    pol = xr.DataArray(pol)
+
+    tmp_dims_f_GHz = f_GHz.dims
+    tmp_dims_pol = pol.dims
 
-    f_GHz = np.asarray(f_GHz)
+    if tmp_dims_f_GHz != tmp_dims_pol:
+        raise ValueError("Frequency and polarization must have identical "
```

Comment:
This is not yet covered by a test. Please add one simple test for that.

Reply:
Should be fixed already by commit 7440056. This [code coverage report](https://github.com/pycomlink/pycomlink/pull/141#issuecomment-1803819904] is three hours old right now.